Flaky integration test: TestMiniCluster_CustomAZRGPreservation (fixed sleep races membership re-convergence)

## Summary

`TestMiniCluster_CustomAZRGPreservation` in `tests/integration` is **flaky**. It failed in CI (observed on the PR #177 "Component Integration Test" job) with:

```
mini_cluster_test.go:949: Should discover all 3 nodes after restart
    expected: 3
    actual  : 2
mini_cluster_test.go:958: node1 AZ should be preserved after restart
    expected: "us-west-2"
    actual  : ""
mini_cluster_test.go:959: node1 RG should be preserved after restart
    expected: "rg-standard"
    actual  : ""
--- FAIL: TestMiniCluster_CustomAZRGPreservation (15.82s)
```

The failure is unrelated to the PR it failed on — it's a pre-existing timing issue in this test.

## Root cause: fixed `time.Sleep` is too short for membership re-convergence

The test leaves node1, restarts it, then waits a fixed `time.Sleep(3 * time.Second)` before asserting the client's discovery sees all 3 nodes with node1's AZ/RG preserved.

But re-joining the memberlist cluster after a restart is asynchronous. The reconcile loop in `server/service.go` (`monitorAndJoinSeeds`) uses a backoff of up to `maxBackoff = 10 * time.Second` (the CI logs show `nextCheck=10s`), so the restarted node + gossip propagation to the client can take well over 3s. The CI logs confirm the cluster reached `newMemberCount=3` right around the moment the assertion ran — the client just hadn't received the update yet.

The empty AZ/RG is a side effect, not metadata loss: when only 2 nodes are discovered, `restartedMeta["node1"]` is a **missing map key** that returns the zero value `""`. The initial pre-restart assertions (3 nodes + correct AZ/RG) pass, and node0/node2 remain correct after restart — only the just-restarted node1 is briefly absent. So this is a convergence-timing race, not a regression in the AZ/RG-preservation feature.

## Why it's clearly a test bug

Every other membership test in this same file already waits for convergence with `require.Eventually` / `assert.Eventually` (e.g. `mini_cluster_test.go:205`, `:270`). This test uniquely used a fixed `time.Sleep(3s)`, which deviates from that established pattern.

## Proposed fix (test-only)

Replace the fixed `time.Sleep(3s)` + one-shot assertions after restart with `require.Eventually(...)` (30s timeout, 500ms poll) that polls `discovery.GetAllServers()` until all 3 nodes are present **and** node1's AZ/RG have re-propagated — matching the convention used by the sibling tests.

--- a/tests/integration/mini_cluster_test.go
+++ b/tests/integration/mini_cluster_test.go
@@ -942,16 +942,31 @@ func TestMiniCluster_CustomAZRGPreservation(t *testing.T) {
 	t.Logf("Restarting node 1...")
 	_, err = cluster.RestartNode(t, 1, cluster.GetSeedList())
 	assert.NoError(t, err)
-	time.Sleep(3 * time.Second)
 
-	// Verify the restarted node has preserved AZ/RG via discovery
-	updatedServers := discovery.GetAllServers()
-	assert.Equal(t, 3, len(updatedServers), "Should discover all 3 nodes after restart")
-
+	// Wait for the client to rediscover all 3 nodes (including the restarted
+	// node1) with node1's AZ/RG re-propagated via gossip. The async seed-join
+	// reconcile loop runs on a backoff of up to 10s, so a fixed sleep is racy
+	// (a 2-node snapshot also makes node1's AZ/RG read as "" from a missing map
+	// key). Poll until membership has actually converged.
 	restartedMeta := make(map[string][2]string)
-	for _, srv := range updatedServers {
-		restartedMeta[srv.NodeId] = [2]string{srv.Az, srv.ResourceGroup}
-		t.Logf("After restart: %s -> AZ=%s, RG=%s", srv.NodeId, srv.Az, srv.ResourceGroup)
+	require.Eventually(t, func() bool {
+		updatedServers := discovery.GetAllServers()
+		if len(updatedServers) != 3 {
+			return false
+		}
+		m := make(map[string][2]string)
+		for _, srv := range updatedServers {
+			m[srv.NodeId] = [2]string{srv.Az, srv.ResourceGroup}
+		}
+		if m["node1"][0] != "us-west-2" || m["node1"][1] != "rg-standard" {
+			return false
+		}
+		restartedMeta = m
+		return true
+	}, 30*time.Second, 500*time.Millisecond, "Client should rediscover all 3 nodes with node1's AZ/RG preserved after restart")
+
+	for nodeID, meta := range restartedMeta {
+		t.Logf("After restart: %s -> AZ=%s, RG=%s", nodeID, meta[0], meta[1])
 	}
 
 	// Verify AZ/RG preserved after restart

